ares.h

```diff
@@ -368,8 +368,8 @@ struct iovec;
 struct ares_socket_functions {
    ares_socket_t(*asocket)(int, int, int, void *);
    int(*aclose)(ares_socket_t, void *);
-   int(*aconnect)(ares_socket_t, const struct sockaddr *, socklen_t, void *);
-   ssize_t(*arecvfrom)(ares_socket_t, void *, size_t, int, struct sockaddr *, socklen_t *, void *);
+   int(*aconnect)(ares_socket_t, const struct sockaddr *, ares_socklen_t, void *);
+   ssize_t(*arecvfrom)(ares_socket_t, void *, size_t, int, struct sockaddr *, ares_socklen_t *, void *);
    ssize_t(*asendv)(ares_socket_t, const struct iovec *, int, void *);
```

Comment:
This was a regression introduced on Jan 31 2017 by https://github.com/c-ares/c-ares/commit/c32103e1126d7100360f2079b580239fe83f4488  so I don't think there should be any discussion about ABI breakage.

But yes, ares_socklen_t *is* socklen_t on systems that have it.  ares_socklen_t exists strictly because not all systems have socklen_t and according to the rest of the c-ares code, should always be used instead of directly using socklen_t.